### DISCUSSION TITLE
How to export data from Fastlane to next step in GitHub Actions

### DISCUSSION THREAD

**Discussion by Dwite** [community/community#27168] [Actions]
<p>Hello,</p>
<p>I have a GitHub Actions job that contains 2 steps:</p>
<ol>
<li>Fastlane</li>
<li>Print message</li>
</ol>
<p>In Fastlane I’m getting some data and trying to expose it as ENV variable to be available for the next GA step</p>
<p>Fastlane part that I’ve tried:</p>
<pre><code class="lang-auto">echo(message:"VERSION_NAME=#{name} &gt;&gt; $GITHUB_ENV")
ENV["VERSION_NAME"] = name
echo(message:"::set-env name=VERSION_NAME::#{name}")
</code></pre>
<p>GH Part that I use</p>
<pre><code class="lang-auto">- name: Build with Fastlane
  run: fastlane code
- name: Print the message
   run: echo "Build for ${{ env.VERSION_NAME }} is ready for testing."
</code></pre>
<p>Any ideas how to share data I got inside Fastlane between steps?</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="Dwite" data-post="1" data-topic="152547">
<div class="title">
<div class="quote-controls"></div>
 Dwite:</div>
<blockquote>
<p><code>echo(message:"VERSION_NAME=#{name} &gt;&gt; $GITHUB_ENV")</code></p>
</blockquote>
</aside>
<p>I’m not familiar with Fastlane, but this looks like you’re <em>printing</em> most (minus the <code>echo</code>) of the <em>shell</em> command that would work.</p>
<p>You need to write the variable in the form <code>NAME=VALUE</code> to the file named by the <code>GITHUB_ENV</code> environment variable, however file access works in Fastlane.</p>

**Comment by Dwite** [comment]
<p>Thank you for pointing that out. I’ve replaced the echo with sh that executes the command as script, but the result is mostly the same<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/2/c2dc78892cf91f6070fd8fc41dc90f31129ff1bd.png" data-download-href="/uploads/short-url/rNOZ7Gr4efCPE1NsvTFrvkhSqdf.png?dl=1" title="Снимок экрана 2020-12-29 в 16.21.48" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/4738859/181102140-9d724b7d-cdfa-4c64-8f02-ce7eb841b7c7.png" alt="Снимок экрана 2020-12-29 в 16.21.48" data-base62-sha1="rNOZ7Gr4efCPE1NsvTFrvkhSqdf" width="344" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/2/c2dc78892cf91f6070fd8fc41dc90f31129ff1bd_2_344x500.png, https://aws1.discourse-cdn.com/github/original/3X/c/2/c2dc78892cf91f6070fd8fc41dc90f31129ff1bd.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/2/c2dc78892cf91f6070fd8fc41dc90f31129ff1bd.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/2/c2dc78892cf91f6070fd8fc41dc90f31129ff1bd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Снимок экрана 2020-12-29 в 16.21.48</span><span class="informations">500×726 54.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment, marked as answer]
<p>The <code>echo</code> part of the shell command seems to be missing.</p>

**Comment by Dwite** [comment]
<p>Thank you very much for the help, indeed this result will work:</p>
<pre><code class="lang-auto">sh("echo VERSION_NAME=#{name} &gt;&gt; $GITHUB_ENV")
sh("echo VERSION_CODE=#{code} &gt;&gt; $GITHUB_ENV")
</code></pre>
<pre><code class="lang-auto">- name: Print the message
   run: echo "Build for ${{ env.VERSION_NAME }}(${{ env.VERSION_CODE }}) is ready for testing."
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/a/aa051cf9dec20d1075cdbfef0657a7ac87db3ba7.png" data-download-href="/uploads/short-url/og47KeqZCSPGFmAENTXDTDYN5B5.png?dl=1" title="Снимок экрана 2020-12-29 в 16.54.20" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/4738859/181102142-5b453f1f-a52f-4532-9d29-65842b9af34c.png" alt="Снимок экрана 2020-12-29 в 16.54.20" data-base62-sha1="og47KeqZCSPGFmAENTXDTDYN5B5" width="527" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/a/aa051cf9dec20d1075cdbfef0657a7ac87db3ba7_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Снимок экрана 2020-12-29 в 16.54.20</span><span class="informations">555×526 47.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by salvatoreboemia** [comment]
Hi @Dwite 

I tried the steps above but I want to add the version to the artifact name 

name: Myartifact_${{ env.VERSION }}

but not work

I need to run something before? (run command or other?

thanks
